train_cnn_lstm_enhanced.py

```
@@ -0,0 +1,831 @@
+        dataset = TensorDataset(X_tensor, y_tensor)
+        batch_size = self.training_config.get("batch_size", 32)
+        
+        return DataLoader(dataset, batch_size=batch_size, shuffle=shuffle)
+
+    def _train_epoch(
+        self,
+        dataloader: DataLoader,
+        optimizer: optim.Optimizer,
+        criterion: nn.Module,
+        epoch: int,
+    ) -> tuple[float, Dict[str, float]]:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)